python/ray/scripts/scripts.py

```diff
@@ -193,13 +193,19 @@ def cli(logging_level, logging_format):
     "--temp-dir",
     default=None,
     help="manually specify the root temporary dir of the Ray process")
+@click.option(
+    "-f",
+    "--config-file",
```

Comment:
--internal-config?

Also why a file and not a string directly?

Reply:
Thanks for the review; I was following the discussion here https://github.com/ray-project/ray/pull/3130#issuecomment-433619966 

Ideally, all of the parameters in `RayConfig` would be accessible here, so maybe the string would become too long? Although I suspect usually you wouldn't modify too many parameters.

@chuxi had some opinions...

Reply:
Typically you want as much configuration in the command line as possible, to avoid confusion about what the defaults are and where the config file is located. The limit is quite large, for example on my machine:

$ getconf ARG_MAX
2097152

So that's 2 MB of configs.

Reply:
A couple things:
- I like `--internal-config` because this should NOT be used by users. We should include a stronger message in the help string saying `"Do NOT use this. This is for debugging and development purposes only."` If people run into issues, they may try to solve their problems by tweaking the config values, but that will not solve their issues and it will be very frustrating.
- I like putting things in a string instead of a config file. However @chuxi and others have expressed strong preferences for config files. One option is to just use a string or something simple in this PR and we can change it later.

Reply:
Given the way that we're using this (that is, programatically in tests), it's going to be much easier to work with strings than with files.